ADBDEV-4726-65 Fix null-check for funcexpr

dnskvlnk opened the pull request:
Fix null-check for funcexpr

ExecMakeTableFunctionResult dereferences funcexpr before null-check making
segfault possible. This patch adds an assertion

Commit:
Fix null-check for funcexpr

ExecMakeTableFunctionResult dereferences funcexpr before null-check making
segfault possible. This patch moves dereferencing after null-check was made

RekGRpth commented on src/backend/executor/execQual.c:
```diff
@@ -2294,6 +2290,9 @@ ExecMakeTableFunctionResult(ExprState *funcexpr,
 		FuncExprState *fcache = (FuncExprState *) funcexpr;
 		ExprDoneCond argDone;
 
+		funcrettype = exprType((Node *) funcexpr->expr);
```
funcexpr also is dereferenced [below](https://github.com/arenadata/gpdb/pull/761/files#diff-924cc2c9527218d9cb8b9606e381bd58ff3a1f74fbf20f65b95c0cf1b525790eR2401)

RekGRpth commented on src/backend/executor/execQual.c:
```diff
@@ -2256,10 +2256,6 @@ ExecMakeTableFunctionResult(ExprState *funcexpr,
 
 	callerContext = CurrentMemoryContext;
 
-	funcrettype = exprType((Node *) funcexpr->expr);
```
I think you should have just replaced the condition `funcexpr &&` with an assert before the dereference here.

dnskvlnk replied:
Done

Commit:
Fix null-check for funcexpr

ExecMakeTableFunctionResult dereferences funcexpr before null-check making
segfault possible. This patch adds an assertion
